From bc5cb34d01546f5fa482d91af4edcd92b001d98c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Sat, 7 Jan 2023 19:27:30 +0000
Subject: [PATCH 06/42] registrychunks: get rid of internal
 GST_REGISTRY_CHUNK_FLAG_MALLOC

Not actually needed.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/3695>
---
 gst/gstregistrychunks.c | 7 ++-----
 gst/gstregistrychunks.h | 4 +---
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/gst/gstregistrychunks.c b/gst/gstregistrychunks.c
index b8a88958..24b01029 100644
--- a/gst/gstregistrychunks.c
+++ b/gst/gstregistrychunks.c
@@ -116,10 +116,7 @@ void
 _priv_gst_registry_chunk_free (GstRegistryChunk * chunk)
 {
   if (!(chunk->flags & GST_REGISTRY_CHUNK_FLAG_CONST)) {
-    if ((chunk->flags & GST_REGISTRY_CHUNK_FLAG_MALLOC))
-      g_free (chunk->data);
-    else
-      g_free (chunk->data);
+    g_free (chunk->data);
   }
   g_free (chunk);
 }
@@ -165,7 +162,7 @@ gst_registry_chunks_save_string (GList ** list, gchar * str)
   chunk = g_new (GstRegistryChunk, 1);
   chunk->data = str;
   chunk->size = strlen ((gchar *) chunk->data) + 1;
-  chunk->flags = GST_REGISTRY_CHUNK_FLAG_MALLOC;
+  chunk->flags = GST_REGISTRY_CHUNK_FLAG_NONE;
   chunk->align = FALSE;
   *list = g_list_prepend (*list, chunk);
   return TRUE;
diff --git a/gst/gstregistrychunks.h b/gst/gstregistrychunks.h
index 927f5803..9121e5fd 100644
--- a/gst/gstregistrychunks.h
+++ b/gst/gstregistrychunks.h
@@ -28,13 +28,11 @@
 
 /*
  * we reference strings directly from the plugins and in this case set CONST to
- * avoid freeing them. If g_free() should be used, the MALLOC flag is set,
- * otherwise g_free() will also be used. (FIXME: don't need MALLOC flag any more)
+ * avoid freeing them.
  */
 enum {
   GST_REGISTRY_CHUNK_FLAG_NONE = 0,
   GST_REGISTRY_CHUNK_FLAG_CONST = 1,
-  GST_REGISTRY_CHUNK_FLAG_MALLOC = 2,
 };
 
 /*
-- 
2.40.1

